File: backend/src/main/java/com/paperai/model/entity/AgentMessage.java
package com.paperai.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * Agent 消息实体
 *
 * @author: ch
 * @date 2026年05月11日
 */
@Data
@TableName("agent_message")
public class AgentMessage {

    /** 消息ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 关联论文ID */
    private Long paperId;

    /** 关联任务ID */
    private Long taskId;

    /** 发送者角色 */
    private String senderRole;

    /** 接收者角色 */
    private String receiverRole;

    /** 消息类型 */
    private String messageType;

    /** 消息内容 */
    private String content;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: backend-py/app/models/agent_message.py
from datetime import datetime
from sqlalchemy import String, Text, BigInteger, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base


class AgentMessage(Base):
    __tablename__ = "agent_message"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    paper_id: Mapped[int | None] = mapped_column(BigInteger)
    task_id: Mapped[int | None] = mapped_column(BigInteger)
    sender_role: Mapped[str] = mapped_column(String(30), nullable=False)
    receiver_role: Mapped[str | None] = mapped_column(String(30))
    message_type: Mapped[str] = mapped_column(String(30), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
